data/items.js

```diff
@@ -396,7 +398,7 @@ exports.BattleItems = {
 			basePower: 30,
 		},
 		onUpdate: function (pokemon) {
-			if (pokemon.hp <= pokemon.maxhp / 2) {
+			if (pokemon.hp <= pokemon.maxhp / 2 && pokemon.lastEffect && pokemon.lastEffect.id !== 'confused') {
```

Comment:
Could `lastEffect` ever be `null` here? Like say you just switched in and your HP is already in range. Maybe return early like you've done for the rest of the Berry checks just to be extra safe.

Reply:
It could, probably, so yeah probably better to move it outside.